fleet_vehicle_stock/tests/test_fleet_vehicle.py

```diff
@@ -0,0 +1,113 @@
+                "type": "product",
+                "fleet_vehicle_model_id": self.vehicle_model.id,
+                "tracking": "serial",
+            }
+        )
+
+        lot1 = self.env["stock.production.lot"].create(
+            {
+                "name": "serial1",
+                "product_id": product1.id,
+                "company_id": self.env.user.company_id.id,
+            }
+        )
+
+        lot2 = self.env["stock.production.lot"].create(
+            {
+                "name": "serial2",
+                "product_id": product1.id,
+                "company_id": self.env.user.company_id.id,
+            }
+        )
+
+        self.env["stock.quant"].create(
+            {
+                "product_id": product1.id,
+                "location_id": self.stock_location.id,
+                "quantity": 1.0,
+                "lot_id": lot1.id,
+            }
+        )
+
+        self.vehicle = self.vehicle.create(
+            {
+                "model_id": self.vehicle_model.id,
+                "product_id": product1.id,
+                "lot_id": lot1.id,
+                "current_stock_location_id": self.stock_location.id,
+            }
+        )
+
+        self.vehicle_without_quant = self.vehicle.create(
+            {
+                "model_id": self.vehicle_model.id,
+                "product_id": product1.id,
+                "lot_id": lot2.id,
+            }
+        )
+
+        self.vehicle_without_lot_id = self.vehicle.create(
+            {"model_id": self.vehicle_model.id, "product_id": product1.id}
+        )
+
+        self.product2 = self.product.create(
+            {
+                "name": "Product B",
+                "type": "product",
+                "fleet_vehicle_model_id": self.vehicle_model.id,
+                "tracking": "serial",
+            }
+        )
+
+    def test_onchange_product(self):
+        vehicle = self.vehicle
+        vehicle._onchange_product()
+        self.assertFalse(vehicle.current_stock_location_id)
+
+    def test_onchange_model(self):
+        vehicle = self.vehicle
+        vehicle._onchange_model()
+        self.assertTrue(vehicle.product_id == vehicle.model_id.product_id)
+
+    def test_compute_current_stock_loc_id(self):
+        vehicle = self.vehicle
+        vehicle._compute_current_stock_loc_id()
+        self.assertTrue(vehicle.current_stock_location_id == self.stock_location)
+
+        vehicle_without_quant = self.vehicle_without_quant
+        vehicle_without_quant._compute_current_stock_loc_id()
+        self.assertTrue(vehicle_without_quant.current_stock_location_id.id is False)
+
+        vehicle_without_lot_id = self.vehicle_without_lot_id
+        vehicle_without_lot_id._compute_current_stock_loc_id()
+        self.assertTrue(vehicle_without_lot_id.current_stock_location_id.id is False)
+
+    def test_inverse_fleet_vehicle_model_id(self):
+        product2 = self.product2
+        product2._inverse_fleet_vehicle_model_id()
```

Comment:
You shouldn't call inverse methods directly, but rather test writing on the field that should trigger it.
Here you want to test that the variant's fields are set when you write on the product template's field